m4/pbs_decl_ppoll.m4

```diff
@@ -48,6 +48,7 @@ AC_DEFUN([PBS_AC_DECL_PPOLL],
      AC_MSG_CHECKING(whether ppoll API is supported)
       AC_TRY_RUN(
 [
+#define _GNU_SOURCE
```

Comment:
We have avoided the use of _GNU_SOURCE within PBS for portability. Is there another way to detect the presence of ppoll() without using _GNU_SOURCE?

Reply:
Per the ppoll man pages: 
#define _GNU_SOURCE         /* See feature_test_macros(7) */
#include <poll.h>
int ppoll(struct pollfd *fds, nfds_t nfds,
        const struct timespec *timeout_ts, const sigset_t *sigmask);
.. poll() conforms to POSIX.1-2001.	 ppoll() is Linux specific

So, I think we need the _GNU_SOURCE enabled for ppoll() to work.

Reply:
Per the ppoll man pages:
#define _GNU_SOURCE /* See feature_test_macros(7) */
#include <poll.h>
int ppoll(struct pollfd *fds, nfds_t nfds,
const struct timespec *timeout_ts, const sigset_t *sigmask);
.. poll() conforms to POSIX.1-2001.	ppoll() is Linux specific

So, I think we need the _GNU_SOURCE enabled for ppoll() to work.

Reply:
In that case, I think we need to use AC_GNU_SOURCE to turn _GNU_SOURCE on everywhere. Otherwise, we could end up with inconsistencies between files where _GNU_SOURCE is not defined. Please refer to this page... https://xaizek.github.io/2013-10-16/autoconf-how-to-define-gnu-source-correctly/

Reply:
Turns out, AC_GNU_SOURCE is obsolete, please see https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Obsolete-Macros.html. The recommendation seems to be to use AC_USE_SYSTEM_EXTENSIONS.

Also, I have to call AC_USE_SYSTEM_EXTENSIONS up above in configure.ac, much before any AC_COMPILE_IFELSE is called from the other macros, else we get errors/warnings as follows:
libtoolize: copying file `m4/ltversion.m4'
libtoolize: copying file `m4/lt~obsolete.m4'
configure.ac:283: warning: AC_COMPILE_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
../../lib/autoconf/specific.m4:368: AC_USE_SYSTEM_EXTENSIONS is expanded from...
configure.ac:283: the top level
configure.ac:283: warning: AC_RUN_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
../../lib/autoconf/specific.m4:368: AC_USE_SYSTEM_EXTENSIONS is expanded from...
configure.ac:283: the top level

Please check